processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -399,7 +402,7 @@ private Assignment assignToPlainViaAdder(Type sourceType, Assignment rightHandSi
                 result = new AdderWrapper(
                     result,
                     method.getThrownTypes(),
-                    getSourceRef(),
+                    getSourceRHS().getSourceReference(),
                     sourceType
```

Comment:
can be optimized... mouth source type and source reference are now present in Freemarker templates.